semantic_cache/semantic_caching.py

```diff
@@ -0,0 +1,188 @@
+    ) -> Optional[List[Generation]]:
+        """
+        Converts the JSON string to generations and returns them.
+        It is called by Langchain to check if the generations are in the cache.
+
+        Args:
+            prompt (str): The prompt to lookup in the cache.
+            llm_string (Optional[str]): Optional string for LLM context.
+
+        Returns:
+            Optional[List[Generation]]:
+            If response is found in the cache; otherwise, None.
+        """
+        result = self.get(prompt)
+        return self._loads_generations(result) if result else None
+
+    def update(
+        self,
+        prompt: str,
+        llm_string: Optional[str] = None,
+        result: Optional[str] = None,
+    ) -> None:
+        """
+        Converts the generations to a JSON string and stores it in the cache.
+        It is called by Langchain to upload the generations to the cache.
+
+        Args:
+            prompt (str): The prompt to update in the cache.
+            llm_string (Optional[str]): Optional string for LLM context.
+            result (Optional[str]): The result to store in the cache.
+        """
+        self.set(prompt, self._dumps_generations(result))
+
+    def set(self, key: Union[str, List[str]], data: Union[str, List[str]]) -> None:
+        """
+        Sets the key and value in the cache.
+
+        Args:
+            key (Union[str, List[str]]): The key(s) to set in the cache.
+            data (Union[str, List[str]]): The value(s) to associate with the key(s).
+        """
+        if isinstance(key, list) and isinstance(data, list):
+            for i in range(len(key)):
+                currrent_key = key[i]
+                self.index.upsert(
+                    [(self._hash_key(currrent_key), currrent_key, {"data": data[i]})]
+                )
+        else:
+            self.index.upsert([(self._hash_key(key), key, {"data": data})])
+
+    def delete(self, key: str) -> None:
+        """
+        Deletes the key from the cache.
+
+        Args:
+            key (str): The key to delete from the cache.
+        """
+        self.index.delete([self._hash_key(key)])
+
+    def bulk_delete(self, keys: List[str]) -> None:
+        """
+        Deletes multiple keys from the cache.
+
+        Args:
+            keys (List[str]): The keys to delete from the cache.
+        """
+        for key in keys:
+            self.delete(key)
+
+    def flush(self) -> None:
+        """
+        Resets the cache, removing all keys and values.
+        """
+        self.index.reset()
+
+    def _query_key(self, key: str):
+        """
+        Queries the cache for the key.
+
+        Args:
+            key (str): The key to query in the cache.
+
+        Returns:
+            The response from the cache query.
+        """
+        response = self.index.query(data=key, top_k=1, include_metadata=True)
+        return response[0] if response else None
+
+    def _is_2d_list(self, lst) -> bool:
```

Comment:
Is this used anywhere?

Reply:
No, I forgot to delete it. It is redundant now